configure.ac

```diff
@@ -443,21 +443,34 @@ if test "x$with_dl" != "xno"; then
   AC_CHECK_LIB(dl, dlopen)
 fi
 
+AC_DEFUN([LIBATOMIC_CHECKER],[
+  AC_LINK_IFELSE([
+    AC_LANG_SOURCE([[
+      #include <atomic>
+      struct LockRequired { char data[256]; };
+      int
+      main() {
+          return std::atomic<LockRequired>{}.is_lock_free();
+      }
+  ]])],[
+    AC_MSG_RESULT(yes)
+    libatomic_checker_result="yes"],[
+    AC_MSG_RESULT(no)
+    libatomic_checker_result="no"
+])])
+
 ## check for atomics library before anything that might need it
 SQUID_STATE_SAVE(LIBATOMIC)
-AC_MSG_CHECKING(library containing __atomic_is_lock_free)
-LIBS="-latomic"
-AC_LINK_IFELSE(
-  [AC_LANG_SOURCE([[
-    int main(void) {
-        char lockRequired[256];
-        return __atomic_is_lock_free(sizeof(lockRequired),&lockRequired);
-    }
-  ]])],
-  [AC_MSG_RESULT(-latomic)
-   ATOMICLIB='-latomic'],
-  [AC_MSG_RESULT(no)]
-)
+AC_MSG_CHECKING(whether linking without -latomic works)
+LIBATOMIC_CHECKER()
```

Comment:
If it is not too difficult, can we move `AC_MSG_CHECKING` inside `LIBATOMIC_CHECKER`, adding a single required `checkingMsg` parameter? This would allow us to visually pair `AC_MSG_CHECKING` with `AC_MSG_RESULT`. The two dependent calls currently live in very different contexts.

Reply:
Done at af1e979.